### PR TITLE
Frontend to create and train custom LUInet models

### PR DESCRIPTION
**Pull request by gcampax**
This PR adds a simple UI to create custom LUInet models that target a subset of Thingpedia, and train them.

This is part of #33 and #205

### CODE REVIEW

File: util/credit_system.js [hunk 1]
```diff
@@ -22,6 +27,23 @@ module.exports = {
...
+
+        // check this so we can change the numbers without changing the calling code
+        if (cost === 0)
+            return;
+
+        const credits = await orgModel.getCredits(dbClient, orgId);
+        if (!(credits >= cost)) {
+            throw new ForbiddenError(req.ngettext(
+                "You do not have enough credits to complete this operation, you need at least %d.",
+                "You do not have enough credits to complete this operation, you need at least %d.", cost)
```
Comment (rayslxu) [inline util/credit_system.js]: Add a link to the explanation or the current credits in dev console?

**Comment by gcampax** [comment]
Thanks for the review.
The vast majority of training time depends on the number of gradient updates, which is fixed at 50k for all models currently. While it is possible that the same model on a smaller dataset converges more quickly, I don't know whether that is the case, and we don't have hyperparameter options to make a smaller model - nor we really want them, because I don't want people do to hyperparameter tuning on our infrastructure.
The models are not stored in RDB, only the source datasets (paraphrase and templates). These are small (order of 10-20k sentences), so I don't foresee it as an issue. Storing the derived/augmented datasets would be problematic, but that's why the other PR stopped doing that.

As for storing the datasets, currently they are simple TSV files on disk. PyTorch does not have a standard example format like TF.
TSV is somewhat space efficient compared to non-compressed, non-numericalized formats, because it has minimal overhead.
decanlp loads the TSV files and then creates a pytorch preprocessed file (really, a pickle file) which is numericalized, and that's more space and memory efficient. Numericalization depends on the vocabulary though, so I don't think it's meaningful to store numericalized datasets.
